### ISSUE TITLE
[New build - DEFAULT] 22w03a: Trap Door and Blast Door

### ISSUE THREAD

**Issue by tsunamayo** [tsunamayo/Starship-EVO#4480]
Blast door and Trap door implementation have been changed.
They are now using children entity, which means you can paint frame and door separately and apply decals on the moving part.
As collision are still deactivated for convenience it is not possible to add bricks on them.

Hotfixes:
#4456 Ship in same squadron can start targeting friendly ship in case of friendly fire.
- Opening several time the mission board is duplicating the last ship entry.
- Bounty Hunting ambush rate was 100%
#4475 Damage and repair on system applied incorrectly.
#4461 Thruster texture scaling incorrect.
#3420 #4457 Trapdoor hitbox issues.
#4463 Symmetry grid placement is incorrect at large scale.
#4479 Warp Jump fail to finish in some cases.

**Comment by tsunamayo** [comment]
Build is up! I got sick of the complexity of the previous blast door implementation so I am now simply using children entities.
It is less optimized but as those are much much faster since all the rendering + physics refactoring so it should be okay (please tell me if you have some FPS on large ships). 
And you can decorate them, as I noticed quite a few people were doing it already (except decals were staying in place) so it should be better.

**Comment by Uncle-Ulty** [comment]
It's a good opportunity to ask something important. Before the neovox update, CE was something with a high impact on performance. Now you change the default doors and make each one of them add a single CE. So my question is: how much does a CE impact performance nowadays?

Why am I asking this... Sometimes I want to build a round engine, that uses several CE (like the one below). But since I don't know the computational impact, I've been using only bricks to build round engines. 

There are also some situations in which I use leaned or curved surfaces, and I build them exclusively with bricks, which results in a higher count of bricks that could be reduced a lot if I had applied some CE on rotators or hinges.

So, that's the current conflict in my mind when I start to build something... I'd love to know the performance impact of bricks vs CE, and maybe some sort of equivalence ratio concerning performance impact.

![image](https://user-images.githubusercontent.com/73253472/149875472-43f50d3b-5dc6-4926-8584-5c73b46353d9.png)

**Comment by tsunamayo** [comment]
@Uncle-Ulty well the impact is much much less compared to what it used to be a long time ago, but there is still an impact. Especially if you have plenty, it might start to lag (I could try to optimize a few things that if that become an issue). I cant give you a brick equivalent because it completely depend on what you are looking for, there GFX performance, collision performance and a tons of other metrics, but I would say it is better to stick with bricks even if you use more. Maybe not if you are to use 100x more indeed, but I doubt it is the case.
Are you mounting your lightcruise thrusters on CE too? Because you are gonna not like to have to pipe those when fuel will be used by lightcruise thrusters...

**Comment by Uncle-Ulty** [comment]
> Are you mounting your lightcruise thrusters on CE too? Because you are gonna not like to have to pipe those when fuel will be used by lightcruise thrusters...

No, all my system bricks are on the main body. Currently, I use CE for details and shapes only. But in the past, I used several CE, in a very inefficient way, which I try to avoid nowadays. I'd rather share the same CE with several objects, for instance, in a recent cargo ship I put all six landing gear on the same CE.
